test/rm.js

```diff
@@ -57,6 +57,17 @@ test('invalid option', t => {
   t.is(result.stderr, 'rm: option not recognized: @');
 });
 
+test('remove symbolic link to a dir without -r fails', t => {
```

Comment:
This statement is only correct because of the trailing slash:

```sh
$ rm link_to_a_dir/ # fails
$ rm link_to_a_dir # succeeds (it removes the link itself, does not affect dir or contents
```

Please clarify this in the test name, and please include a more verbose comment explaining the significance of the trailing slash.

---

Also, we have an existing test which handles the case with no trailing slash and no `-r` (yay), however that test case needs an assertion. Currently, it asserts that the destination dir still exists, but it should instead assert that the *contents* of that destination dir still exist. The destination **always** survives an `rm`. If you can, please fix that test too.

Reply:
Other ways we might test link-handling:

 * `-r`, link to a dir, trailing slash (delete contents and the link)
 * `-rf`, link to a dir, trailing slash (this is already implemented, but I mention it here for completeness)
 * `-r`, link to a dir, no trailing slash (deletes only the link, not the destination, nor the destination's contents)
 * `-rf`, link to a dir, no trailing slash (same as above, probably not necessary to copy this out)

Reply:
I've tested the first one and I think the behavior is a little different (linux 4.4.0, bash 4.3.48):

- `-r`, link to a dir, trailing slash: `rm: cannot remove 'dir/': Not a directory`

Can you tell me if you get the same behavior in bash?

Reply:
Yes, I see the same behavior on my machine.

Reply:
Also, `rm -rf linkToDir/` will exit without error, but will remove neither the link nor the target.